### ISSUE TITLE
ERROR: Dictionary::_unref on closing Godot

### ISSUE THREAD

**Issue by DmitriySalnikov** [GameAnalytics/GA-SDK-GODOT#8]
Windows 10 64bit
26986be43d1a80b98e6ba945f1a65238824be697
https://github.com/godotengine/godot/commit/adf2dabbde23fdd89803871e26dd1d33ca20bf77 and every other 3.2.4 commits, may be all 3.2 versions idk

![image](https://user-images.githubusercontent.com/7782218/99075156-0b693d80-25ca-11eb-9b07-5378565927c9.png)

That happens on unregistering GDScript module only with integrated GA SDK
![image](https://user-images.githubusercontent.com/7782218/99080499-73bc1d00-25d2-11eb-89a2-5a207ea9d52e.png)

**Comment by the1schwartz** [comment]
Ok I will have a look at what is going wrong. So it is happening when you close the Godot editor and having the gameanalytics module enabed, right?

**Comment by DmitriySalnikov** [comment]
Yes, it is.
You can just open and close project manager with enabled GA.
![ZmsQsvRP8C](https://user-images.githubusercontent.com/7782218/99274808-d106d700-283b-11eb-81a4-35ef82da7191.gif)

Maybe the problem is in Godot itself but I only see this problem with GA enabled

**Comment by the1schwartz** [comment]
Ok thanks I will have a look to try to figure out what needs to be changed to fix this

**Comment by the1schwartz** [comment]
Is it happening with previous version of the GameAnalytics SDK module?

**Comment by the1schwartz** [comment]
Also I am wondering how the GA module could affect unregistering GDScript module?

**Comment by the1schwartz** [comment]
Should we report it on Godot github issues?

**Comment by DmitriySalnikov** [comment]
> Is it happening with previous version of the GameAnalytics SDK module?

same in 1.2.0, 1.1.0, 1.0.4, 1.0.2 

---
After some commits, Godot 3.2 branch now displays this new error
```
Godot Engine v3.2.4.beta.custom_build.06389f705 - https://godotengine.org
OpenGL ES 3.0 Renderer: GeForce GTX 970/PCIe/SSE2
OpenGL ES Batching: ON

ERROR: Engine::Singleton::Singleton: A class intended to be used as a singleton must *not* inherit from Reference.
   At: core\engine.cpp:245
ERROR: Dictionary::_unref: Condition "!_p" is true.
   At: core\dictionary.cpp:185
```

so the error is here
https://github.com/GameAnalytics/GA-SDK-GODOT/blob/26986be43d1a80b98e6ba945f1a65238824be697/gameanalytics/GameAnalytics.h#L6-L8

**Comment by DmitriySalnikov** [comment]
After replacing `Reference` with `Object`, the problem disappeared.

**Comment by the1schwartz** [comment]
Ok thanks I will add that fix to the next version

**Comment by the1schwartz** [comment]
This has been fixed now in the latest version (v1.2.2)
